com.unity.visualeffectgraph/Editor/Data/VFXDataParticle.cs

```diff
@@ -744,6 +760,8 @@ public bool NeedsGlobalSort()
 
                 var context = m_Contexts[i];
                 var contextData = contextToCompiledData[context];
+                contextData.indexInSystemDesc = outSystemDescs.Count;
+                contextToCompiledData[context] = contextData;
```

Comment:
contextToCompiledData must not be modified inside FillDesc, it is supposed to be immutable within those calls. parameters that are written to uses out prefix

Reply:
See https://github.com/Unity-Technologies/Graphics/pull/4457#issuecomment-839768246